[CI] S3BlobContainerRetriesTests testReadBlobWithReadTimeouts failing

**Build Scans:**
- [elasticsearch-intake #34261 / part1](https://gradle-enterprise.elastic.co/s/p44fbgagbmfym)
- [elasticsearch-periodic #11842 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/korcq5ibr2kno)

**Reproduction Line:**
```
./gradlew ":modules:repository-s3:test" --tests "org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests.testReadBlobWithReadTimeouts" -Dtests.seed=9D151A2843ECE20A -Dtests.locale=it-SM -Dtests.timezone=Asia/Chongqing -Druntime.java=25
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testReadBlobWithReadTimeouts'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<5> but was:<3>
```

**Issue Reasons:**
- [9.3] 2 failures in test testReadBlobWithReadTimeouts (4.8% fail rate in 42 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -357,9 +357,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.inference.loadingservice.ModelLoadingServiceTests
   method: testExpiredModelsAreEvictedBeforeLoading
   issue: https://github.com/elastic/elasticsearch/issues/141589
-- class: org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests
-  method: testReadBlobWithReadTimeouts
-  issue: https://github.com/elastic/elasticsearch/issues/141960
 - class: org.elasticsearch.search.aggregations.metrics.TopHitsIT
   method: testMixedSortFieldTypes
   issue: https://github.com/elastic/elasticsearch/issues/142077

